src/main/resources/org/taktik/icure/asyncdao/impl/js/patient/By_hcparty_address_postalcode_housenumber.js

```diff
@@ -0,0 +1,52 @@
+map = function (doc) {
+
+	var emit_normalized_substrings = function (k, txt, postalCode, houseNumber, docId, latin_map) {
+		var r = txt.toLowerCase().replace(/[^A-Za-z0-9]/g, function (a) {
+			return latin_map[a] || ""
+		});
+		for (var i = 0; i <= r.length - 3; i++) {
+			emit([k, (r.substr(i, r.length - i)), postalCode, houseNumber], docId);
+		}
+	};
+
+	var emit_patients_by_address = function (hcparty, doc, latin_map) {
+		doc.addresses.forEach(function (address) {
+			emit_normalized_substrings(hcparty, ((address.street || '') + '|' + (address.city || '')).replace(new RegExp('\\s', 'g'), '').replace(new RegExp('\\W', 'g'), '').toLowerCase(), address.postalCode, address.houseNumber, doc._id, latin_map);
```

Comment:
Why changing the content of the `searchString` ? You want to add the postalCode & the houseNumber to filter more presicely your solution, ok : But what would prevent the user to put the postal code in the searchString if he wants ? 
==> Keep postalCode & houseNumber in the searchString, to let its signification the same. Otherwise, do a complete new filter, where the `searchString` would be renamed to `street` and `city`.